common/src/main/java/edu/snu/nemo/common/coder/IntDecoder.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2018 Seoul National University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.snu.nemo.common.coder;
+
+import java.io.*;
+
+/**
+ * A {@link Decoder} which is used for an integer.
+ */
+public final class IntDecoder implements Decoder<Integer> {
+
+  /**
+   * A private constructor.
+   */
+  private IntDecoder() {
+    // do nothing.
+  }
+
+  /**
+   * Static initializer of the coder.
+   */
+  public static IntDecoder of() {
+    return new IntDecoder();
```

Comment:
Since `IntDecoder` is stateless, we'd better return statically initialized `IntDecoder` instance rather than creating new instances on every `IntDecoder#of` incarnation.